XML解析からJSON解析に変更

kobayashota opened the pull request:
## 概要

#3 の課題に関するプルリクエスト

## 変更点

- Yahoo!RSSからニュース記事を取得後、XML解析→Qiita API V2から記事を検索後、JSON解析
- テーブルビューセルの背景画像を削除
- テーブルビューセルの背景色をsystemgreenに設定

## キャプチャ映像

https://user-images.githubusercontent.com/67745523/116024523-59aa5e80-a689-11eb-813f-b121ae0d369a.mp4

## 懸念点

- 固定のキーワードでの検索のみ閲覧可能であること（ワードを指定して検索が必要？）
- セルの余白が広すぎる（表示するセルの数を増やすべきか）

など

kobayashota committed:
XML解析からJSON解析に変更

kobayashota committed:
Delete Article.swift

必要ないファイル

kobayashota committed:
#9 修正, pod update 実行

kozukaaaa commented on Swift5IntroApp1/Controller/BaseViewController.swift:
```diff
@@ -57,61 +56,39 @@ class BaseViewController: SegementSlideDefaultViewController{
+            urlString = "https://qiita.com/api/v2/items?query=\(words[0])"
 
         }
 
```
こういう情報はvcに書くと「ビジネスロジックの肥大化」になるので別定義がいいですね

kobayashota replied:
この場合、Modelフォルダに分けるべきですか？

kozukaaaa replied:
基本そうですね

kozukaaaa commented on Swift5IntroApp1/Controller/NewsPageViewController.swift:
```diff
@@ -9,64 +9,32 @@
+import SwiftyJSON
 
-class NewsPageViewController: UITableViewController, SegementSlideContentScrollViewDelegate,XMLParserDelegate{
+class NewsPageViewController: UITableViewController, SegementSlideContentScrollViewDelegate{
```
```SegementSlideContentScrollViewDelegate {```がいいです

kobayashota replied:
クラス宣言時の余白をつける #9
をPushしました 
https://github.com/kobashou06/practice-xml-parse/commit/30e831c0327612d029a772b8f86a5a801aaa223c

kobayashota committed:
Merge remote-tracking branch 'refs/remotes/origin/my-patch-1'

kozukaaaa commented on Swift5IntroApp1/Controller/NewsPageViewController.swift:
```diff
@@ -9,64 +9,32 @@
         
+        super.viewDidLoad()
+        request()
+        print(self.jsonDataArray)
```
print残っちゃってますね

kobayashota replied:
消してきます！

kobayashota replied:
#12 に登録しました

kozukaaaa commented on Swift5IntroApp1/Controller/NewsPageViewController.swift:
```diff
@@ -79,141 +47,84 @@ class NewsPageViewController: UITableViewController, SegementSlideContentScrollV
-        self.currentElementName = nil
-    
-    }
-    
```
xmlとjsonで別クラスでswitchできるようにしとくとスマートかなと感じました

kobayashota replied:
XMLに関する処理を全て消してコミットした後に、それを実感しました、、

kozukaaaa replied:
ここのをコピペして再構成すればいいんじゃないですかね？

kozukaaaa commented on Swift5IntroApp1/Controller/NewsPageViewController.swift:
```diff
@@ -79,141 +47,84 @@ class NewsPageViewController: UITableViewController, SegementSlideContentScrollV
-        let newsItem = newsItems[indexPath.row]
-        UserDefaults.standard.set(newsItem.url, forKey: "url")
+        let article = jsonDataArray[indexPath.row]
+        UserDefaults.standard.set(article.url, forKey: "url")
```
UserDefaultsで管理してるのはなんでなんですかね？

kobayashota replied:
教材のやり方をそのまま使いました、、
教材でもこれはUserDefaultsは普段使わないと言っていたので、別の手段に切り替え必要と感じてます

kozukaaaa replied:
単純にvcに値渡して実装してみてください

kozukaaaa commented on Swift5IntroApp1/Controller/NewsPageViewController.swift:
```diff
@@ -79,141 +47,84 @@ class NewsPageViewController: UITableViewController, SegementSlideContentScrollV
+                    let json:JSON = try JSON(data: response.data!)
+                    var totalHitCount = json.count
+                    
+                    if totalHitCount > 50{
```
50件までしか表示できない...？ 🤔

kobayashota replied:
これも教材で件数を制限していたのを流用しています、、 😢 
件数制限は必要ないですか？

kozukaaaa replied:
件数制限では無く、maxだったらfetchしないがいいですね
私もちょうどそれやっているので
https://retailai.atlassian.net/jira/software/projects/TPAY/boards/80?assignee=5e7afa037276650c2d53bcfb&selectedIssue=TPAY-754

kobayashota committed:
#12 修正

kobayashota committed:
WebViewが閉じない問題 #7 修正
## 変更点
- Modalでの画面遷移を廃止
- NavigationControl のPushで画面遷移
- 前の画面に戻れないバグが解消された

kobayashota committed:
WebViewが閉じない問題 #7 追加修正１
